public/src/Helpers.ts

```diff
@@ -28,8 +29,7 @@ export enum ConnStatus {
 const months = ["January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"]
 
 /** Returns a string with a prettier format for a deadline */
-export const getFormattedTime = (deadline_string: string): string => {
-    const deadline = new Date(deadline_string)
+export const getFormattedTime = (deadline: Date): string => {
```

Comment:
We can replace this `deadline: Date` argument and do the following:
```go
export const getFormattedTime = (timestamp: Timestamp | undefined): string => {
    if (!timestamp) {
        return "N/A"
    }
    const deadline = timestamp.toDate()
    return `${deadline.getDate()} ${months[deadline.getMonth()]} ${deadline.getFullYear()} ${deadline.getHours()}:${deadline.getMinutes() < 10 ? "0" : ""}${deadline.getMinutes()}`
}
```
Then we can simplify the use of all uses of this function; we always do this (or equal alternative):
```typescript
assignment.deadline ? getFormattedTime(assignment.deadline.toDate()) : "N/A"
```
Which would become:
```typescript
getFormattedTime(assignment.deadline)
```